utils/sysfs/sysfs_test.go

```diff
@@ -0,0 +1,128 @@
+
+import (
+	"github.com/stretchr/testify/assert"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+func TestGetNodes(t *testing.T) {
+	//overwrite global variable
+	nodeDir = "./testdata/"
+
+	sysFs := NewRealSysFs()
+	nodesDirs, err := sysFs.GetNodesPaths()
+	assert.Nil(t, err)
+	assert.Equal(t, 2, len(nodesDirs))
+	assert.Contains(t, nodesDirs, "testdata/node0")
+	assert.Contains(t, nodesDirs, "testdata/node1")
+}
+
+func TestGetNodesWithNonExistingDir(t *testing.T) {
+	//overwrite global variable
+	nodeDir = "./testdata/NonExistingDir/"
+
+	sysFs := NewRealSysFs()
+	nodesDirs, err := sysFs.GetNodesPaths()
+	assert.Nil(t, err)
+	assert.Equal(t, 0, len(nodesDirs))
+}
+
+func TestGetCPUsPaths(t *testing.T) {
+	sysFs := NewRealSysFs()
+	cpuDirs, err := sysFs.GetCPUsPaths("./testdata/node0")
+	assert.Nil(t, err)
+	assert.Equal(t, 2, len(cpuDirs))
+	assert.Contains(t, cpuDirs, "testdata/node0/cpu0")
+	assert.Contains(t, cpuDirs, "testdata/node0/cpu1")
+}
+
+func TestGetCPUsPathsFromNodeWithoutCPU(t *testing.T) {
+	sysFs := NewRealSysFs()
+	cpuDirs, err := sysFs.GetCPUsPaths("./testdata/node1")
+	assert.Nil(t, err)
+	assert.Equal(t, 0, len(cpuDirs))
+}
+
+func TestGetCoreID(t *testing.T) {
+	sysFs := NewRealSysFs()
+	rawCoreID, err := sysFs.GetCoreID("./testdata/node0/cpu0")
+	assert.Nil(t, err)
+
+	coreID, err := strconv.Atoi(strings.TrimSpace(string(rawCoreID)))
+	assert.Nil(t, err)
+	assert.Equal(t, 0, coreID)
+}
+
+func TestGetCoreIDWhenFileIsMissing(t *testing.T) {
+	sysFs := NewRealSysFs()
+	rawCoreID, err := sysFs.GetCoreID("./testdata/node0/cpu1")
+	assert.NotNil(t, err)
+	assert.Equal(t, []byte(nil), rawCoreID)
+}
+
+func TestGetMemInfo(t *testing.T) {
+	sysFs := NewRealSysFs()
+	memInfo, err := sysFs.GetMemInfo("./testdata/node0")
+	assert.Nil(t, err)
+	assert.Equal(t, "Node 0 MemTotal:       32817192 kB", strings.TrimSpace(string(memInfo)))
+}
+
+func TestGetMemInfoWhenFileIsMissing(t *testing.T) {
+	sysFs := NewRealSysFs()
+	memInfo, err := sysFs.GetMemInfo("./testdata/node1")
+	assert.NotNil(t, err)
+	assert.Equal(t, []byte(nil), memInfo)
+}
+
+func TestGetHugePagesInfo(t *testing.T) {
+	sysFs := NewRealSysFs()
+	hugePages, err := sysFs.GetHugePagesInfo("./testdata/node0/hugepages")
+	assert.Nil(t, err)
+	assert.Equal(t, 2, len(hugePages))
+
+	expectedHugePages := []string{"hugepages-1048576kB", "hugepages-2048kB"}
+	for _, hugePage := range hugePages {
+		assert.Contains(t, expectedHugePages, hugePage.Name())
+	}
+}
+
+func TestGetHugePagesInfoWhenDirIsMissing(t *testing.T) {
+	sysFs := NewRealSysFs()
+	hugePages, err := sysFs.GetHugePagesInfo("./testdata/node1/hugepages")
+	assert.NotNil(t, err)
+	assert.Equal(t, []os.FileInfo([]os.FileInfo(nil)), hugePages)
+}
+
+func TestGetHugePagesNr(t *testing.T) {
+	sysFs := NewRealSysFs()
+	rawHugePageNr, err := sysFs.GetHugePagesNr("./testdata/node0/hugepages/", "hugepages-1048576kB")
+	assert.Nil(t, err)
+
+	hugePageNr, err := strconv.Atoi(strings.TrimSpace(string(rawHugePageNr)))
```

Comment:
See my comment about trimming above. Value returned from `GetHugePagesNr()` must be a number, doesn't it?